src/connector.ts

```diff
@@ -1,64 +1,117 @@
-import { Signer } from 'ethers'
-import { ZeroWalletConnectorOptions, ZeroWalletSigner } from './types'
-import { ZeroWalletProvider } from './provider'
+import { ethers, Signer } from 'ethers'
+import { ZeroWalletConnectorOptions } from './types'
+import { ZeroWalletProvider, ZeroWalletSigner } from './provider'
 import { Chain, Connector, ConnectorData } from 'wagmi'
 
 export class ZeroWalletConnector extends Connector<ZeroWalletProvider, ZeroWalletConnectorOptions, ZeroWalletSigner> {
-	readonly id = 'zero-wallet'
-	readonly name = 'Zero Wallet'
+    readonly id = 'zero-wallet'
+    readonly name = 'Zero Wallet'
 
-	private provider: ZeroWalletProvider
+    private signer: ZeroWalletSigner
 
-	constructor(config: { chains?: Chain[]; options: ZeroWalletConnectorOptions }) {
-		super(config)
-		this.provider = new ZeroWalletProvider(config.options.jsonRpcProviderUrl)
-	}
+    constructor(config: { chains?: Chain[]; options: ZeroWalletConnectorOptions }) {
+        super(config)
+        this.signer = new ZeroWalletSigner(config.options.jsonRpcProviderUrl)
+    }
 
-	get ready() {
-		return true
-	}
+    get ready() {
+        return true
+    }
 
-	async connect(): Promise<Required<ConnectorData>> {
-		throw new Error('not implemented')
-	}
+    async connect(): Promise<Required<ConnectorData>> {
 
-	async disconnect(): Promise<void> {
-		throw new Error('not implemented')
-	}
+        if(localStorage.getItem('ZeroWalletConnected') === 'true') {
+            throw new Error("Already connected!");
+        }
 
-	async getAccount(): Promise<string> {
-		throw new Error('not implemented')
-	}
+        const provider = await this.getProvider()
+        if (!provider) throw new Error("Provider not found");
 
-	async getChainId(): Promise<number> {
-		throw new Error('not implemented')
-	}
+        if (provider.on) {
+            provider.on('accountsChanged', this.onAccountsChanged)
+            provider.on('chainChanged', this.onChainChanged)
+            provider.on('disconnect', this.onDisconnect)
+        }
 
-	async getProvider(): Promise<ZeroWalletProvider> {
-		return this.provider
-	}
+        const privateKey = localStorage.getItem('zeroWalletPrivateKey');
```

Comment:
Can we please abstract this away?

```
storage = StorageFactory.create(platform) ; //platform = "browser | react-native" for now
storage.get('zerowalletPrivateKey')
```

This is so that by changing the platform parameter (or auto detecting) while initializing the connector, we can change this param and it just works.

Please note we'll need this connector to work with react native too